BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java

```diff
@@ -0,0 +1,256 @@
+            //Get connection object from database based on id
+            this.currentConnection = getConnection(id);
+
+            //create opcua connection to that machine
+            try {
+                client = getOpcUaClient(currentConnection.getIp());
+            } catch (InterruptedException e) {
+                logger.log(Level.SEVERE, e.getMessage(), e);
+                Thread.currentThread().interrupt();
+            } catch (ExecutionException | UaException e) {
+                logger.log(Level.SEVERE, e.getMessage(), e);
+            }
+        }
+    }
+
+    private OpcUaClient getOpcUaClient(String ipAddress) throws InterruptedException, ExecutionException, UaException {
+        //get all endpoints from the machine
+        List<EndpointDescription> endpoints = DiscoveryClient.getEndpoints(ipAddress).get();
+
+        String[] ipAddressArray = ipAddress.split(":");
+        EndpointDescription configPoint = EndpointUtil.updateUrl(endpoints.get(0), ipAddressArray[1].substring(2), Integer.parseInt(ipAddressArray[2]));
+
+        //loading endpoints into configuration
+        OpcUaClientConfigBuilder cfg = new OpcUaClientConfigBuilder();
+        cfg.setEndpoint(configPoint);
+
+        //setting up machine client with config
+        OpcUaClient opcUaClient = OpcUaClient.create(cfg.build());
+
+        //connecting machine
+        opcUaClient.connect().get();
+        return opcUaClient;
+    }
+
+    private void sendChanges() {
+        try {
+            NodeId cmdChangeRequest = CommandNodes.EXECUTE_MACHINE_COMMAND.nodeId;
+            client.writeValue(cmdChangeRequest, DataValue.valueOnly(new Variant(true))).get();
+        } catch (InterruptedException | ExecutionException e) {
+            logger.log(Level.SEVERE, e.getMessage(), e);
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    /**
+     * Takes a command and sends it to the desired machine
+     *
+     * @param command      The {@code command} that should be sent to the machine
+     * @param connectionID The ID of the machine that should receive the {@code command}
+     * @return Returns {@code true} if the control {@code command} was successfully sent, returns {@code false} if the control {@code command} failed
+     */
+    @Override
+    public boolean controlMachine(Command command, String connectionID) {
+        connectToMachine(connectionID);
+
+        try {
+            NodeId cntrlCmd = CommandNodes.SET_MACHINE_COMMAND.nodeId;
+            client.writeValue(cntrlCmd, DataValue.valueOnly(new Variant(command.label))).get();
+            sendChanges();
+            return true;
+        } catch (InterruptedException e) {
+            logger.log(Level.SEVERE, e.getMessage(), e);
+            Thread.currentThread().interrupt();
+            return false;
+        } catch (ExecutionException e) {
+            logger.log(Level.SEVERE, e.getMessage(), e);
+            return false;
```

Comment:
Should we interrupt the thread with this exception as well?

Reply:
The addition of the thread.interrupt(), was purely because of a code smell.